src/main/java/org/jboss/pnc/cleaner/auth/AuthenticationException.java

```diff
@@ -0,0 +1,38 @@
+/**
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2014 Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.pnc.cleaner.auth;
+
+/**
+ * This exception is thrown when there's something wrong with the authentication mechanism.
+ *
+ * It will result in 500 status code for requests.
+ *
+ *
+ * Author: Michal Szynkiewicz, michal.l.szynkiewicz@gmail.com
+ * Date: 3/23/16
+ * Time: 12:32 PM
+ */
+public class AuthenticationException extends RuntimeException {
```

Comment:
@pkocandr Why not to add dependency on Orch rather then copy the code?
      <dependency>
        <groupId>org.jboss.pnc</groupId>
        <artifactId>auth</artifactId>
        <version>${pnc.version}</version>
      </dependency>

Reply:
I was considering that, but
* it pulls in a lot of transitive dependencies that are really not needed
* it seemed the CDI setup won't fit our use case and might need a lot of playing around, so I copied for now only the necessary pieces

I wanted to avoid starting up the whole PNC in the cleaner... But I have to admit that I already needed to depend on spi because of 1 class in REST API is needed from there (BuildCoordinationStatus - which is a bug obviously) so maybe this point is already lost :-)

Reply:
What about to split up the auth module to auth-model and auth-smth? We could add only the auth-model here.

What problems do you see with CDI? After reading the Quarkus docs, I think we can use it without issues. I definitively plan it for the TemporaryBuildsCleaner

Reply:
Yeah, I was probably too lazy to do that in both 2.0 and 1.8, resolve all the dependencies accross modules correctly etc, but anyway, it seems it needs to be done. But not sure if "auth-model" is a good name either. Maybe auth-core and something else.

By CDI issues I meant something like a class instantiates and tries to inject a dependency, which doesn't have an implementation on classpath, it might need some configuration which we don't have in cleaner etc.

Reply:
auth-model might not be the best name, but I think it could be split sometimes to be able to reuse it other project. I'd go for it. Copying the code is smth I'd like to avoid.